File: services/reservation/src/main/java/com/example/iter/reservation/scheduler/RentalExpirationScheduler.java
package com.example.iter.reservation.scheduler;

import com.example.iter.reservation.service.RentalService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

// 결제(confirm)를 30분 안에 완료하지 않은 PENDING 요청을 주기적으로 자동 취소해서 선점을 풀어준다.
@Slf4j
@Component
@RequiredArgsConstructor
public class RentalExpirationScheduler {

    private final RentalService rentalService;

    @Scheduled(fixedRate = 60_000)
    public void expirePendingRentals() {
        int expiredCount = rentalService.expirePendingRentals();
        if (expiredCount > 0) {
            log.info("30분 초과 미결제 대여요청 {}건 자동 취소", expiredCount);
        }
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/scheduler/RentalExpirationScheduler.kt
package com.example.iter.reservation.scheduler

import com.example.iter.reservation.service.RentalService
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

// 결제(confirm)를 30분 안에 완료하지 않은 PENDING 요청을 주기적으로 자동 취소해서 선점을 풀어준다.
@Component
class RentalExpirationScheduler(
    private val rentalService: RentalService,
) {

    @Scheduled(fixedRate = 60_000)
    fun expirePendingRentals() {
        val expiredCount = rentalService.expirePendingRentals()
        if (expiredCount > 0) {
            log.info("30분 초과 미결제 대여요청 {}건 자동 취소", expiredCount)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(RentalExpirationScheduler::class.java)
    }
}
